contracts/colony/ColonyExpenditure.sol

```diff
@@ -316,10 +315,11 @@ contract ColonyExpenditure is ColonyStorage {
 
       // Validate payout modifier
       if (offset == 2) {
-        if (!ColonyAuthority(address(authority)).hasUserRole(msgSender(), 1, uint8(ColonyDataTypes.ColonyRole.Root))){
-          require(int256(uint256(_value)) <= 0, "colony-expenditure-bad-payout-modifier");
-        }
-        require(int256(uint256(_value)) >= MIN_PAYOUT_MODIFIER, "colony-expenditure-bad-payout-modifier");
+        require(
+          (int256(uint256(_value)) <= 0 || IColony(address(this)).hasUserRole(msgSender(), 1, ColonyRole.Root)) &&
```

Comment:
You've changed from querying the Authority to querying the Colony here. Is that deliberate?

Reply:
I'm also not really a fan of this refactoring in general, I think this is a lot harder to follow.

Reply:
It was deliberate, since we already have a `hasUserRole` in the colony interface. It seemed like an abstraction violation to go to the authority here.

In general, I don't like putting `require` statements inside of conditionals, since that can more easily be expressed as a two-condition `require`. If nothing else is in the branch except a `require`, I would say just put the condition inside the `require`.

Reply:
> It was deliberate, since we already have a hasUserRole in the colony interface. It seemed like an abstraction violation to go to the authority here.

You're right, it's just gassier. I guess it depends which 'incorrect' side we want to err on.

> In general, I don't like putting require statements inside of conditionals, since that can more easily be expressed as a two-condition require. If nothing else is in the branch except a require, I would say just put the condition inside the require.

I fear that by 'more easily', you mean 'in fewer characters', which isn't the goal when we're not code-golfing. All other things being equal, we should go for the easiest to understand with all the nested bracket stacked up in the require, I really don't think it is.

I actually don't think either of us are right here, and it should be expressed with a conditional but should be how you've done it in the require (i.e. checking `value` first, and only doing the external call if necessary, which is much cheaper).

Reply:
True, it would be better if `hasUserRole` was in `ColonyStorage`, since I think this is the first time we're using the function for an internal call. I could accept a call directly to the authority on the grounds that `hasUserRole` was meant for external calls only and so this usage doesn't represent an abstraction violation.

Reply:
Regarding the `require`, I made my point in an earlier comment. When scanning the contract, having multiple conditions inside of a `require` tells the reader that all of this logic ultimately is used to validate the caller. The fact that the contents of the `require` are slightly more obtuse is a trade-off, but one which I argue makes the overall contract easier to understand. If the reader needs to understand the specifics of the `require`, they can then make the extra effort to parse the contents. But if not, then the more dense version makes the contract as a whole easier to interpret.